File: PythonDS/stack.py
#Stack ds 

class Stack:
    def __init__(self):
        self.stack = []

    def push(self, item):
        self.stack.append(item)

    def pop(self):
        if self.is_empty():
            return None
        return self.stack.pop()

    def peek(self):
        if self.is_empty():
            return None
        return self.stack[-1]

    def is_empty(self):
        return len(self.stack) == 0

    def size(self):
        return len(self.stack)


# Example usage:
my_stack = Stack()
my_stack.push(5)
my_stack.push(10)
my_stack.push(15)

print(my_stack.pop())  # Output: 15
print(my_stack.peek())  # Output: 10
print(my_stack.is_empty())  # Output: False
print(my_stack.size())  # Output: 2

File: SwiftDS/stack.swift
//Stack in swift

import Foundation

struct Stack<T> {
    private var elements = [T]()
    
    mutating func push(_ element: T) {
        elements.append(element)
    }
    
    mutating func pop() -> T? {
        return elements.popLast()
    }
    
    func peek() -> T? {
        return elements.last
    }
    
    func isEmpty() -> Bool {
        return elements.isEmpty
    }
    
    func count() -> Int {
        return elements.count
    }
}

//Test
var stack = Stack<Int>()
stack.push(1)
stack.push(27)
stack.push(3)

//stack: 1, 27, 3
//top: 3
stack.count() // 3
stack.peek() // 3
stack.pop() // 3
stack.pop() // 27
stack.count() // 1
stack.isEmpty() // false
stack.pop() // 1
stack.isEmpty() // true
